Cellular: Document all possible error codes for cellular API.

jarvte opened the pull request:
### Description

Document all possible error codes for cellular API.

Internal ref to defect: IOTCELL-882

No code changes, only documentation update.

@AnttiKauppila @AnotherButler please review.

### Pull request type

    [X] Fix
    [ ] Refactor
    [ ] Target update
    [ ] Feature
    [ ] Breaking change

Commit:
Cellular: Document all possible error codes for cellular API.

AnttiKauppila commented on features/cellular/framework/API/CellularNetwork.h:
```diff
@@ -312,45 +324,54 @@ class CellularNetwork : public NetworkInterface {
-     *  @return     zero on success
+     *  @param rat           Radio access technology
+     *  @return              NSAPI_ERROR_OK on success
+     *                       NSAPI_ERROR_UNSUPPORTED if the given opAct is RAT_UNKNOWN, inhering target class
```
inhering?

Commit:
Cellular: review fixes for doxygen changes of return values.

AnttiKauppila reviewed (changes requested):
We cannot document inherited class' error codes, we only document "default" use case.

AnttiKauppila reviewed (approved):
LGTM

jarvte commented:
@AnotherButler can you review?

Commit:
Copy edit CellularInformation.h

Copy edit file for parallelism and active voice.

AnotherButler commented:
Query: Do any of these APIs need their own user API pages in the handbook?

Commit:
Copy edit CellularNetwork.h

Copy edit file for parallelism and active voice.

Commit:
Copy edit CellularPower.h

Copy edit file for parallelism, and fix typo

Commit:
Copy edit CellularSIM.h

Copy edit for parallelism, and fix typo and capitalization from before these changes.

Commit:
Copy edit CellularSMS.h

Copy edit file for parallelism, consistent capitalization and minor grammar nits.

jarvte commented:
@AnotherButler there is no need for API pages in handbook. Lorawan didn't have that either.

0xc0170 commented:
/morph build

AnotherButler commented:
@jarvte Thanks for confirming 👍
